docs/_data/nav_docs.yml

```diff
@@ -32,6 +32,8 @@
       title: JSX In Depth
     - id: typechecking-with-proptypes
       title: Typechecking With PropTypes
+    - id: test-utils
+      title: Test Utilities
```

Comment:
This is a reference so it should probably go in the "reference" rather than "advanced guides".
Can be last item there, after synthetic events.